src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -1090,10 +1103,35 @@ internal bool TryGetNodeInfo(string name, out CustomNodeInfo info)
                         //    input_var_name : type
                         if (parameters != null && parameters.Count() > inputReceiverData)
                         {
-                            var typeName = parameters[inputReceiverData].DisplayTypeName;
-                            if (!string.IsNullOrEmpty(typeName))
+                            var port = inputReceiverNode.InPorts[inputReceiverData];
+                            var typedParameter = parameters[inputReceiverData];
+                            // initially set the type name to the full type name
+                            // then try to shorten it.
+                            if (!string.IsNullOrEmpty(typedParameter.Type.Name))
                             {
-                                node.InputSymbol += " : " + typeName;
+                                try
+                                {
+
+                                    var typedNode = new TypedIdentifierNode
+                                    {
+                                        Name = port.Name,
+                                        Value = port.Name,
+                                        datatype = typedParameter.Type,
+                                        TypeAlias = typedParameter.Type.Name
+                                    };
+                                 
+                                    NodeToCodeCompiler.ReplaceWithShortestQualifiedName(
+                                        classTable,
+                                        new List<AssociativeNode> { typedNode },
+                                        currentWorkspace.ElementResolver);
+
+                                    node.InputSymbol = $"{typedNode.Value} :{typedNode.TypeAlias}";
+                                }
+                                catch(Exception e)
+                                {
+                                    node.InputSymbol += ":" + typedParameter.Type.Name;
+                                    this.AsLogger().LogError($"{e.Message}: could not generate a short type name for {typedParameter.Type.Name}");
+                                }
```

Comment:
Does this set `InputSymbol` to the full name even though it is unable to find a short name? If not, it should.

Reply:
yep, thats what it does.